plugins/discovery/discovery_test.go

```diff
@@ -1346,6 +1347,40 @@ func TestReconfigureWithUpdates(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Unexpected error %v", err)
 	}
+
+	// check that not setting persistence_directory doesn't remove boot config
+	updatedBundle = makeDataBundle(13, `
+		{
+			"config": {}
+		}
+	`)
+
+	err = disco.reconfigure(ctx, download.Update{Bundle: updatedBundle})
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+
+	if manager.Config.PersistenceDirectory == nil {
+		t.Fatal("Erased persistence directory configuration")
+	}
+
+	// update persistence directory
+	updatedBundle = makeDataBundle(14, `
+		{
+			"config": {
+				"persistence_directory": "my_bundles"
+			}
+		}
+	`)
+
+	err = disco.reconfigure(ctx, download.Update{Bundle: updatedBundle})
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+
+	if manager.Config.PersistenceDirectory == nil || *manager.Config.PersistenceDirectory != "test" {
```

Comment:
Shouldn't `*manager.Config.PersistenceDirectory` be `my_bundles` 🤔

Reply:
Well good thing the test run caught this https://github.com/open-policy-agent/opa/actions/runs/5400634875/jobs/9809367373?pr=6042#step:6:288 🙂